src/Compilers/Core/Portable/Binding/UseSiteInfo.cs

```diff
@@ -82,6 +85,17 @@ public UseSiteInfo<TAssemblySymbol> AdjustDiagnosticInfo(DiagnosticInfo? diagnos
 
             return this;
         }
+
+        public void MergeDependencies(ref TAssemblySymbol? primaryDependency, ref ImmutableHashSet<TAssemblySymbol>? secondaryDependencies)
+        {
+            secondaryDependencies = (secondaryDependencies ?? ImmutableHashSet<TAssemblySymbol>.Empty).Union(SecondaryDependencies ?? ImmutableHashSet<TAssemblySymbol>.Empty);
+            primaryDependency ??= PrimaryDependency;
+
+            if (!object.Equals(primaryDependency, PrimaryDependency) && PrimaryDependency is object)
+            {
+                secondaryDependencies = secondaryDependencies.Add(PrimaryDependency);
```

Comment:
The docs indicate this will be empty if `DiagnosticInfo` is an error. Should we be conditioning that here?

Reply:
> The docs indicate this will be empty if DiagnosticInfo is an error. Should we be conditioning that here?

I do not think this needs an explicit check because ```PrimaryDependency``` would be null in that case.

---
In reply to: [362669608](https://github.com/dotnet/roslyn/pull/40640#discussion_r362669608) [](ancestors = 362669608)